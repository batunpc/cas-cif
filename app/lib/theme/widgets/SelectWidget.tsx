import { WidgetProps } from "@rjsf/core";
import Dropdown from "@button-inc/bcgov-theme/Dropdown";
import getRequiredLabel from "../utils/getRequiredLabel";

const SelectWidget: React.FunctionComponent<WidgetProps> = (props) => {
  const { id, onChange, options, placeholder, label, required, uiSchema } = props;
  console.log(props)
  return (
    <>
      <label htmlFor={`funding-stream-dropdown-${props.id}`}>
        {getRequiredLabel(label, required)}
      </label>
      <Dropdown
        id={id}
        onChange={(e) => onChange(e.target.value || undefined)}
        placeholder={placeholder}
        size={(uiSchema && uiSchema["bcgov:size"]) || "large"}
        required={required}
      >
        {options.enumOptions.map((opt) => {
          return (
            <option key={opt.value} value={opt.value}>
              {opt.label}
            </option>
          );
        })}
      </Dropdown>
      <style jsx>
        {`
          :global(input) {
            width: 100%;
          }
        `}
      </style>
    </>
  );
};

export default SelectWidget;