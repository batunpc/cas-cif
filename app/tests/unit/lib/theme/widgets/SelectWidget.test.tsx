import React from "react";
import SelectWidget from "lib/theme/widgets/SelectWidget";

import { render, fireEvent, screen } from "@testing-library/react";

describe("The SelectWidget Widget", () => {
  it("Matches the snapshot with no default value set", () => {
    const props: any = {
      id: "1",
      placeholder: "test placeholder",
      onChange: jest.fn(),
      label: "Search",
      value: undefined,
      required: false,
      schema: {
        anyOf: [
          { value: 1, enum: [1], type: "number", title: "Option 1" },
          { value: 2, enum: [2], type: "number", title: "Option 2" },
        ],
      },
    };

    const componentUnderTest = render(<SelectWidget {...props} />);
    expect(componentUnderTest.container).toMatchSnapshot();
  });

  it("selects the first option from default", async () => {
    const props: any = {
      id: "test-dropdown",
      placeholder: "test-placeholder",
      onChange: jest.fn(),
      label: "test-dropdown-label",
      value: undefined,
      required: false,
      schema: {
        anyOf: [
          { value: 1, enum: [1], type: "number", title: "Option 1" },
          { value: 2, enum: [2], type: "number", title: "Option 2" },
        ],
      },
    };

    const selectWidget = render(<SelectWidget {...props} />);
    const dropdown = selectWidget.container.querySelector("#test-dropdown");
    expect(
      screen.getByRole("combobox", { name: "test-dropdown-select" }).value
    ).toEqual("test-placeholder");
    fireEvent.change(dropdown, { target: { value: "1" } });
    expect(
      screen.getByRole("combobox", { name: "test-dropdown-select" }).value
    ).toEqual("1");
  });

  it("Matches the snapshot with a default value set", () => {
    const props: any = {
      id: "1",
      placeholder: "test placeholder",
      onChange: jest.fn(),
      label: "Search",
      value: 1,
      required: false,
      schema: {
        anyOf: [
          { value: 1, enum: [1], type: "number", title: "Option 1" },
          { value: 2, enum: [2], type: "number", title: "Option 2" },
        ],
      },
    };

    const componentUnderTest = render(<SelectWidget {...props} />);

    expect(componentUnderTest.container).toMatchSnapshot();
  });
});
