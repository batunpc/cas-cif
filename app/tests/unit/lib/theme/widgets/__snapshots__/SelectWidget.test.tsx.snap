// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The SelectWidget Widget Matches the snapshot with a default value set 1`] = `
<div>
  <label
    class="jsx-3703916905"
    for="funding-stream-dropdown-1"
  >
    Search
  </label>
  <div
    class="sc-eCImPb dfYLVh pg-select"
  >
    <div
      class="sc-gKclnd kMxstE pg-select-wrapper"
    >
      <select
        aria-label="1-select"
        class="sc-iCfMLu gsBcnF pg-select-input"
        id="1"
        name="1-select"
        placeholder="test placeholder"
      >
        <option
          class="jsx-3703916905"
        >
          test placeholder
        </option>
        <option
          class="jsx-3703916905"
          value="1"
        >
          Option 1
        </option>
        <option
          class="jsx-3703916905"
          value="2"
        >
          Option 2
        </option>
      </select>
    </div>
  </div>
</div>
`;

exports[`The SelectWidget Widget Matches the snapshot with no default value set 1`] = `
<div>
  <label
    class="jsx-3703916905"
    for="funding-stream-dropdown-1"
  >
    Search
  </label>
  <div
    class="sc-eCImPb dfYLVh pg-select"
  >
    <div
      class="sc-gKclnd kMxstE pg-select-wrapper"
    >
      <select
        aria-label="1-select"
        class="sc-iCfMLu gsBcnF pg-select-input"
        id="1"
        name="1-select"
        placeholder="test placeholder"
      >
        <option
          class="jsx-3703916905"
        >
          test placeholder
        </option>
        <option
          class="jsx-3703916905"
          value="1"
        >
          Option 1
        </option>
        <option
          class="jsx-3703916905"
          value="2"
        >
          Option 2
        </option>
      </select>
    </div>
  </div>
</div>
`;
